### PR TITLE
CREATE/DROP plan node cleanup and executors

### PR DESCRIPTION
**Pull request by mbutrovich**
Overview of changes:

- There's a static util class in `execution/sql` that can take in `CREATE` or `DROP` plan nodes along with an `ExecutionContext` and perform the operation. Is this the API we want? Names are undecided and easy to change. I'm not happy with them right now.
- I gutted any fields that weren't required to execute these plan nodes. Note: I did not remove fields that we would eventually want to support (i.e. constraints for CREATE TABLE).  I only removed fields that legitimately seemed redundant.
- Removed `SetFromCreateStatement` and `SetFromDropStatement` methods on plan node Builders. These were used for a few contrived tests, but represent a special case code path that will not be used in the real system. All plan nodes will be generated by going through the parser, binder, and optimizer.
- Removed references to skiplist and ART indexes from `PostgresParser`. I'm not a fan of dead code coming over from Peloton.
- Typing up #635 pretty much convinced me that `IF EXISTS` should not be in physical plan nodes so I removed that field, but that issue exists to discuss it.

### CODE REVIEW

**Comment by mbutrovich** [comment]
While working on this, I realized that the only drop operation that currently cascades in the Catalog is drop database. In an effort to limit the scope of this PR, I'm pushing cascading to the next PR since we'll still need that behavior, particularly for drop namespace for #632.

File: src/include/parser/parser_defs.h [hunk 1]
```diff
@@ -60,8 +60,9 @@ enum class IndexType {
   INVALID = INVALID_TYPE_ID,  // invalid index type
   BWTREE = 1,                 // bwtree
   HASH = 2,                   // hash
-  SKIPLIST = 3,               // skiplist
-  ART = 4,                    // ART
+  // TODO(Matt) are we ever gonna support these again? if not we should remove them
```
Comment (lmwnshn) [inline src/include/parser/parser_defs.h]: We could just go ahead and remove them. If/when someone adds them, they can easily add it here.

File: src/parser/postgresparser.cpp [hunk 2]
```diff
@@ -1314,10 +1314,11 @@ std::unique_ptr<SQLStatement> PostgresParser::CreateIndexTransform(ParseResult *
     index_type = IndexType::BWTREE;
   } else if (strcmp(access_method, "hash") == 0) {
     index_type = IndexType::HASH;
-  } else if (strcmp(access_method, "skiplist") == 0) {
-    index_type = IndexType::SKIPLIST;
-  } else if (strcmp(access_method, "art") == 0) {
-    index_type = IndexType::ART;
+    // TODO(Matt) are we ever gonna support these again? if not we should remove them
```
Comment (lmwnshn) [inline src/parser/postgresparser.cpp]: Can probably be removed.

**Comment by mbutrovich** [comment]
Want to also document that this PR strengthens the asserts in the Catalog and removes some TODOs related to `IF EXISTS` and future expected behavior from the binder. I will open a new issue for that.

**Pull request merged**